### ISSUE TITLE
Does Response() work for anyone?

### ISSUE THREAD

**Issue by juanitogan** [pallets/flask#1799]
I've been trying to stream some data out of the database into a downloadable CSV file and having trouble at every turn.  I've been following what guides I can find.  Response() gives errors and setting headers on make_response() fails to produce a response the browser recognizes as savable (even with `response.headers["Content-Disposition"] = "attachment;filename=study.csv"` ).  I've never gone this direction with data extractions so I can't say I know what to expect with HTTP.

Anyhow, here are the messages from attempts to use Response:
```
   return Response(generate(), mimetype='text/csv')
TypeError: __init__() takes 1 positional argument but 2 were given

    return Response(generate())
TypeError: __init__() takes 1 positional argument but 2 were given

    return Response("1,2,3")
TypeError: __init__() takes 1 positional argument but 2 were given
```

**Comment by RonnyPfannschmidt** [comment]
more context please - without knowing what you do we have no idea what went wrong and the calls look correct on their own, if you use this as stack-overflow replacement, at least provide full context

**Comment by juanitogan** [comment]
I believe I gave too much context instead of too little.  That's why it looks a lot like a StackOverflow question.  Really, the point is Response generates an error and doesn't appear to work... at all.  That's an issue and not an SO question (yet).  But, okay, if you want more code, not much to see other than what I already posted:
```
	test = [("1","2","3"),("4","5","6")]
	def generate():
		for row in test:
			yield ','.join(row) + '\n'
	return Response(generate(), mimetype='text/csv')
```
Pretty much the same as this: [http://flask.pocoo.org/docs/0.10/patterns/streaming/](url)

**Comment by RonnyPfannschmidt** [comment]
i just created a virtualenv with flask and bpython:
```pycon
>>> from flask import Response
>>> test = [("1","2","3"),("4","5","6")]
>>> def generate():
...     for row in test:
...         yield ','.join(row) + '\n'
...     
... 
>>> Response(generate(), mimetype='text/csv')
<Response streamed [200 OK]>
>>> Response(generate(), mimetype='text/csv').data
'1,2,3\n4,5,6\n'
>>> import flask
>>> import sys
>>> sys.version
'2.7.11 (default, Mar 31 2016, 20:46:51) \n[GCC 5.3.1 20151207 (Red Hat 5.3.1-2)]'
>>> flask.__version__
'0.10.1'
```
so please show what you actually do with context that tells is more than correct calls break

**Comment by juanitogan** [comment]
Okay, thank you.  I get the same results when run that way -- thus, I can now see it works in the same venv at least.  It appears I need to look deeper and see why it isn't working through the REST API I'm building.  Running through localhost:5000.  This was the very last API of the project.  Arg.  All else is working.  Probably something very basic I'm missing.

Anyway, in case it means something to someone,  here's the whole proc:
```
@app.route("/api/admin/studies/<int:study_id>/csv", methods=["GET"])
@admin.login_required
def admin_get_csv(study_id):

	test = [("1","2","3"),("4","5","6")]
	def generate():
		for row in test:
			yield ','.join(row) + '\n'

	return Response(generate(), mimetype='text/csv')
```
and the full error stack:
```
127.0.0.1 - - [07/May/2016 03:53:36] "GET /api/admin/studies/4/csv HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/home/Jernigan/.virtualenvs/tlfb/lib/python3.4/site-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/Jernigan/.virtualenvs/tlfb/lib/python3.4/site-packages/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/home/Jernigan/.virtualenvs/tlfb/lib/python3.4/site-packages/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/Jernigan/.virtualenvs/tlfb/lib/python3.4/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/home/Jernigan/.virtualenvs/tlfb/lib/python3.4/site-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/Jernigan/.virtualenvs/tlfb/lib/python3.4/site-packages/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/Jernigan/.virtualenvs/tlfb/lib/python3.4/site-packages/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/Jernigan/.virtualenvs/tlfb/lib/python3.4/site-packages/flask/_compat.py", line 33, in reraise
    raise value
  File "/home/Jernigan/.virtualenvs/tlfb/lib/python3.4/site-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/Jernigan/.virtualenvs/tlfb/lib/python3.4/site-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/Jernigan/.virtualenvs/tlfb/lib/python3.4/site-packages/flask_httpauth.py", line 88, in decorated
    return f(*args, **kwargs)
  File "/home/Jernigan/tlfb/app/adminapi.py", line 403, in admin_get_csv
    return Response(generate(), mimetype='text/csv')
TypeError: __init__() takes 1 positional argument but 2 were given
```

**Comment by RonnyPfannschmidt** [comment]
still showing nothing but code that looks correct, where is your Response type imported from?

**Comment by juanitogan** [comment]
Yeah, I know.  Frustrating.  I have the whole app running and now this on the last day.  Regardless, I appreciate you talking to me.  Thank you.
```
from app import app
from flask import request, abort, g, jsonify, Response, make_response
from flask.ext.jsontools import jsonapi
from flask.ext.httpauth import HTTPTokenAuth
from sqlalchemy.orm import noload, contains_eager
from datetime import date, time, datetime
# JSON encoding.
from app.json_encoder import AlchemyEncoder
app.json_encoder = AlchemyEncoder
```

**Comment by RonnyPfannschmidt** [comment]
can you show the output if `pip freeze` ?

**Comment by juanitogan** [comment]
```
(tlfb) ➜  tlfb git:(master) ✗ pip freeze
alembic==0.8.4
Flask==0.10.1
Flask-HTTPAuth==3.1.2
flask-jsontools==0.1.1.post0
Flask-Migrate==1.7.0
Flask-Script==2.0.5
Flask-SQLAlchemy==2.1
inflection==0.3.1
itsdangerous==0.24
Jinja2==2.8
Mako==1.0.3
MarkupSafe==0.23
passlib==1.6.5
psycopg2==2.6.1
python-editor==0.5
SQLAlchemy==1.0.11
Werkzeug==0.11.3
wheel==0.26.0
```

**Comment by juanitogan** [comment]
and in case you are curious:
```
>>> from flask import Response
>>> test = [("1","2","3"),("4","5","6")]
>>> def generate():
...     for row in test:
...         yield ','.join(row) + '\n'
...
>>> Response(generate(), mimetype='text/csv')
<Response streamed [200 OK]>
>>> Response(generate(), mimetype='text/csv').data
b'1,2,3\n4,5,6\n'
>>> import flask
>>> import sys
>>> sys.version
'3.4.3 (default, May  5 2015, 17:04:32) \n[GCC 4.9.2]'
>>> flask.__version__
'0.10.1'
```

**Comment by juanitogan** [comment]
Oh, duh.  Found it.  I import my models with * and one of them is named Response for the responses table.  I found this after trying `import flask` and `flask.Response` and that worked.  Now, I just need to figure out why the browser isn't triggering the save when it received the response.  But I can take that conversation elsewhere.

Thank you for taking the time.  It helped me even though it was my own dumb oversight.
